src/coreclr/vm/win32threadpool.cpp

```diff
@@ -1,4276 +0,0 @@
-
-SVAL_IMPL(ThreadpoolMgr::ThreadCounter,ThreadpoolMgr,CPThreadCounter);
-
-SVAL_IMPL_INIT(LONG,ThreadpoolMgr,MaxLimitTotalCPThreads,1000);   // = MaxLimitCPThreadsPerCPU * number of CPUS
-SVAL_IMPL(LONG,ThreadpoolMgr,MinLimitTotalCPThreads);
-SVAL_IMPL(LONG,ThreadpoolMgr,MaxFreeCPThreads);                   // = MaxFreeCPThreadsPerCPU * Number of CPUS
-
-// Cacheline aligned, hot variable
-DECLSPEC_ALIGN(MAX_CACHE_LINE_SIZE) SVAL_IMPL(ThreadpoolMgr::ThreadCounter, ThreadpoolMgr, WorkerCounter);
-
-SVAL_IMPL(LONG,ThreadpoolMgr,MinLimitTotalWorkerThreads);          // = MaxLimitCPThreadsPerCPU * number of CPUS
-SVAL_IMPL(LONG,ThreadpoolMgr,MaxLimitTotalWorkerThreads);        // = MaxLimitCPThreadsPerCPU * number of CPUS
-
-SVAL_IMPL(LONG,ThreadpoolMgr,cpuUtilization);
-
-HillClimbing ThreadpoolMgr::HillClimbingInstance;
-
-// Cacheline aligned, 3 hot variables updated in a group
-DECLSPEC_ALIGN(MAX_CACHE_LINE_SIZE) LONG ThreadpoolMgr::PriorCompletedWorkRequests = 0;
-DWORD ThreadpoolMgr::PriorCompletedWorkRequestsTime;
-DWORD ThreadpoolMgr::NextCompletedWorkRequestsTime;
-
-LARGE_INTEGER ThreadpoolMgr::CurrentSampleStartTime;
-
-unsigned int ThreadpoolMgr::WorkerThreadSpinLimit;
-bool ThreadpoolMgr::IsHillClimbingDisabled;
-int ThreadpoolMgr::ThreadAdjustmentInterval;
-
-#define INVALID_HANDLE ((HANDLE) -1)
-#define NEW_THREAD_THRESHOLD            7       // Number of requests outstanding before we start a new thread
-#define CP_THREAD_PENDINGIO_WAIT 5000           // polling interval when thread is retired but has a pending io
-#define GATE_THREAD_DELAY 500 /*milliseconds*/
-#define GATE_THREAD_DELAY_TOLERANCE 50 /*milliseconds*/
-#define DELAY_BETWEEN_SUSPENDS (5000 + GATE_THREAD_DELAY) // time to delay between suspensions
-
-Volatile<LONG> ThreadpoolMgr::Initialization = 0;            // indicator of whether the threadpool is initialized.
-
-bool ThreadpoolMgr::s_usePortableThreadPool = false;
-bool ThreadpoolMgr::s_usePortableThreadPoolForIO = false;
-
-// Cacheline aligned, hot variable
-DECLSPEC_ALIGN(MAX_CACHE_LINE_SIZE) unsigned int ThreadpoolMgr::LastDequeueTime; // used to determine if work items are getting thread starved
-
-SPTR_IMPL(WorkRequest,ThreadpoolMgr,WorkRequestHead);        // Head of work request queue
-SPTR_IMPL(WorkRequest,ThreadpoolMgr,WorkRequestTail);        // Head of work request queue
-
-//unsigned int ThreadpoolMgr::LastCpuSamplingTime=0;      //  last time cpu utilization was sampled by gate thread
-unsigned int ThreadpoolMgr::LastCPThreadCreation=0;     //  last time a completion port thread was created
-unsigned int ThreadpoolMgr::NumberOfProcessors; // = NumberOfWorkerThreads - no. of blocked threads
-
-
-CrstStatic ThreadpoolMgr::WorkerCriticalSection;
-CLREvent * ThreadpoolMgr::RetiredCPWakeupEvent;       // wakeup event for completion port threads
-CrstStatic ThreadpoolMgr::WaitThreadsCriticalSection;
```

Comment:
These two critical sections could be removed from `CrstTypes.def` [here](https://github.com/dotnet/runtime/blob/c4ea2caaa63322da54e289f490ba3bb4820e9303/src/coreclr/inc/CrstTypes.def#L487-L489), [here](https://github.com/dotnet/runtime/blob/c4ea2caaa63322da54e289f490ba3bb4820e9303/src/coreclr/inc/CrstTypes.def#L483-L485), and [here](https://github.com/dotnet/runtime/blob/c4ea2caaa63322da54e289f490ba3bb4820e9303/src/coreclr/inc/CrstTypes.def#L295). Then `crsttypes.h` could be regenerated using `CrstTypeTool.cs`.